src/components/Pressable/Pressable.tsx

```diff
@@ -26,9 +26,29 @@ import { GestureTouchEvent } from '../../handlers/gestureHandlerCommon';
 const DEFAULT_LONG_PRESS_DURATION = 500;
 
 export default function Pressable(props: PressableProps) {
-  const [pressedState, setPressedState] = useState(
-    props.testOnly_pressed ?? false
-  );
+  const {
+    testOnly_pressed,
+    hitSlop,
+    pressRetentionOffset,
+    delayHoverIn,
+    onHoverIn,
+    delayHoverOut,
+    onHoverOut,
+    delayLongPress,
+    unstable_pressDelay,
+    onPress,
+    onPressIn,
+    onPressOut,
+    onLongPress,
+    style,
+    children,
+    android_disableSound,
+    android_ripple,
+    disabled,
+    ...remainingProps
+  } = useMemo(() => props, [props]);
```

Comment:
Do we need to use `useMemo`?

Reply:
I think so, we have some `useStates` which may cause rerenders. If we were to remove this `useMemo`, those props would trigger every `useCallback` to rerender each render as well, redering them useless.

Reply:
I don't think it's necessary, memoization of props passed to the component should be on the user. For example, we cannot correctly memoize event handlers, without knowing what's in their closure (and we don't have access to it).

Adding to that, I don't think it actually does anything here. React uses [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) to compare the dependencies. You're passing an entire `props` object as a dependency, which is created anew for every render and objects are compared based on reference, not value.

Reply:
I see, that makes sense. 
I'll remove all the `useMemo` on `props` instances, all `useCallback` instances which depend on any of the `props`.

Do you think there are any other sensible ways to go about optimising the now non-memoized functions, or should they be left as-is?

Reply:
Other calls to `useCallback` and `useMemo` are fine. I meant specifically this one where you used the entire `props` object in the dependency array.

Reply:
If a user uses `useCallback` to memoize event handlers we want to be able to utilize that and make less updates ourselves.

Reply:
Yes I understand, I kept all the other `useMemo` and `useCallback` instances as all the values they depend on may be primitive values.

Removed prop memoization in [cc3ecdc](https://github.com/software-mansion/react-native-gesture-handler/pull/3039/commits/cc3ecdc486b90e003e9ee1b0a53f58ddc1ded734)